packages/medusa/src/api-v2/admin/campaigns/[id]/promotions/route.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../../types/routing"
+
+import { addOrRemoveCampaignPromotionsWorkflow } from "@medusajs/core-flows"
+import { LinkMethodRequest } from "@medusajs/types/src"
+import { refetchCampaign } from "../../helpers"
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<LinkMethodRequest>,
+  res: MedusaResponse
+) => {
+  const { id } = req.params
+  const { add, remove } = req.validatedBody
+
+  const workflow = addOrRemoveCampaignPromotionsWorkflow(req.scope)
+  const { errors } = await workflow.run({
```

Comment:
very nit and non-blocking:
```suggestion
  const { errors } = await addOrRemoveCampaignPromotionsWorkflow(req.scope).run({
```